### PR TITLE
Upgrade zstd-jni to 1.5.2.2

### PR DESCRIPTION
**Pull request by lic9**
zstd 1.5.2 includes a few improvement (i.e. fix of performance regression introduced in 1.5.0, compression perf improvement, and Huffman improvements: https://github.com/facebook/zstd/releases)
Internal test showed around ~2% overall cpu reduction (decompress dropped from 6% to 4.9%, and compress dropped from 1.9% to 1.6% based on cpu profiling).

Test plan -
Built presto and test internally.

```
== RELEASE NOTES ==

General Changes
* Upgrade zstandard compression to version 1.5.2.2. This can improved about 2% overall CPU with data compressed with zstandard.
```

### CODE REVIEW

**Comment by NikhilCollooru** [comment]
Did we compare the GC activity with the new version ? 
I remember there was a UER miss previously related to zstd version bump. (T71450283)

**Comment by NikhilCollooru** [comment]
Please also add a release note under` General Changes`

File: pom.xml [hunk 1]
```diff
@@ -1929,7 +1929,7 @@
             <dependency>
                 <groupId>com.github.luben</groupId>
                 <artifactId>zstd-jni</artifactId>
-                <version>1.5.0-4</version>
+                <version>1.5.2-2</version>
```
Comment (NikhilCollooru) [inline pom.xml:1932]: Did we compare the GC activity with the new version ?
I remember there was a UER miss previously related to zstd version bump. (T71450283)

Comment (lic9) [reply pom.xml:1932]: Yes, we compared both young GC and old GC, and there were no difference between the 2 versions.

**Review by NikhilCollooru — APPROVED** [review]

**Review by highker — APPROVED** [review]

**Pull request merged**